frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,154 @@
+  const namespace = match.params.ns;
+  const { prometheusTenancyBaseURL } = window.SERVER_FLAGS;
+  const [response, loadError, loading] = useURLPoll<PrometheusRulesResponse>(
+    `${prometheusTenancyBaseURL}/api/v1/rules?namespace=${namespace}`,
+    POLL_DELAY,
+    namespace,
+  );
+  const thanosRules = React.useMemo(
+    () => (!loading && !loadError ? getAlertsAndRules(response?.data).rules : []),
+    [response, loadError, loading],
+  );
+
+  React.useEffect(() => {
+    const sortThanosRules = _.sortBy(thanosRules, (rule) => ruleStateOrder(rule));
+    dispatch(monitoringSetRules('devRules', sortThanosRules));
+  }, [dispatch, thanosRules]);
+
+  const filteredRules = React.useMemo(() => {
+    const filterRulesByStateAndSeverity =
+      alertStateParams || alertSeverityParams
+        ? _.filter(
+            rules,
+            (r) =>
+              _.includes(alertStateParams, r?.state) ||
+              _.includes(alertSeverityParams, r?.labels.severity),
+          )
+        : rules;
+    const filterRulesByName = name
+      ? _.filter(
+          filterRulesByStateAndSeverity,
+          (r) => name && _.includes(_.lowerCase(r.name), _.lowerCase(name)),
+        )
+      : filterRulesByStateAndSeverity;
+    return filterRulesByName;
+  }, [alertSeverityParams, alertStateParams, name, rules]);
+
+  React.useEffect(() => {
+    const tableRows = monitoringAlertRows(filteredRules);
+    _.forEach(tableRows, (tr) => {
+      _.forEach(collapsedRows, (cr) => {
+        if (tr.cells[0].id === cr.cells[0].id) {
+          tr.isOpen = cr.isOpen;
+        }
+      });
+    });
+    setRows(tableRows);
+  }, [filteredRules, collapsedRows]);
+
+  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
+    rows[rowKey].isOpen = isOpen;
+    const collapsedRowsIndex = _.findIndex(
+      collapsedRows,
+      (row) => row.cells[0].id === rows[rowKey].cells[0].id,
+    );
+    if (collapsedRowsIndex === -1) {
+      setCollapsedRows([...collapsedRows, rows[rowKey]]);
+    } else if (collapsedRowsIndex > -1 && isOpen) {
+      collapsedRows.splice(collapsedRowsIndex, 1);
+      setCollapsedRows(collapsedRows);
+    }
+    setRows([...rows]);
+  };
+  const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
+    const sortedRules = _.sortBy(filteredRules, (rule) => {
+      switch (index) {
+        case 1:
+          return rule.name;
+        case 2:
+          return alertSeverityOrder(rule);
+        case 3:
+          return alertStateOrder(rule);
+        default:
+          return 0;
+      }
+    });
+    setSortBy({ index, direction });
+    setRows(
+      monitoringAlertRows(direction === SortByDirection.asc ? sortedRules : sortedRules.reverse()),
+    );
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Alerts</title>
+      </Helmet>
+      <div className="odc-monitoring-alerts">
+        <FilterToolbar
+          rowFilters={alertFilters}
```

Comment:
I'm thinking we can now use `alertsRowFilters` from the admin view directly here.

I think it's fine to change the `filterGroupName` to `Alert State` in the admin view if that's what we want for the dev view.

Reply:
@kyoto In dev perspective, we don't have a default selected filter. Whereas ```alertsRowFilters``` has a default selected filter that is why I reuse only items and reducer from the ```alertsRowFilters```

Reply:
👍